Fix/debug window

ermario opened the pull request:
I added back the debug window for Level 1 and Level 2, I removed some useless functionalities and added new ones (change weapon) that could be useful for testing.

PLEASE, check the prefabs and models to see that I didnt fucked up anything. I had lots of problems with libraries and assets so I did a mess probably. Glasias!

Izenz reviewed (changes requested):
Hey!

When I instantiate the Debug Window prefab onto a scene its script is missing all the references except Tp Pos 1/2/3.
I added the references and tested out a bit tho, here's my opinion.

- Spawn enemy button crashes the game.
- Would be nice to display Atk power and Mov speed values to edit them propperly.
- Atk. Power and Mov. speed can take negative values resulting in weird unwanted behavior, would be wise to clamp after editing.
- Tp Pos values teleport outside the map, may want to fix their values.

Cheers!

baransrc commented on Gameplay/entities/player/PlayerController.cpp:
```diff
@@ -705,6 +705,18 @@ void Hachiko::Scripting::PlayerController::ChangeWeapon(unsigned weapon_idx)
 
+void Hachiko::Scripting::PlayerController::ReloadAmmo(unsigned ammo)
+{
+	if (0 < ammo && ammo <= MAX_AMMO) 
```
```suggestion
	if (ammo <= MAX_AMMO) 
```
No needs to check for smaller than zero as ammo is unsigned.

baransrc commented on Gameplay/entities/player/PlayerController.cpp:
```diff
@@ -705,6 +705,18 @@ void Hachiko::Scripting::PlayerController::ChangeWeapon(unsigned weapon_idx)
+		_ammo_count = ammo;
+		UpdateAmmoUI();
+	}
+	else {
+		HE_LOG("ReloadAmmo: Incorrect AMMO value");
+	}
```
Maybe it's a better idea to add an assert statement here instead of a HE_LOG as no one checks there :eyes:

baransrc reviewed (changes requested):
Thanks for the good contribution. Some small changes, afterwards, we are good to go given that the issues brought up by @Izenz are addressed.

ermario commented:
> Hey!
> 
> When I instantiate the Debug Window prefab onto a scene its script is missing all the references except Tp Pos 1/2/3. I added the references and tested out a bit tho, here's my opinion.
> 
> * Spawn enemy button crashes the game.
> * Would be nice to display Atk power and Mov speed values to edit them propperly.
> * Atk. Power and Mov. speed can take negative values resulting in weird unwanted behavior, would be wise to clamp after editing.
> * Tp Pos values teleport outside the map, may want to fix their values.
> 
> Cheers!

Regarding this, spawn enemy right now can not be fixed because whenever you create a prefab of the enemy, the enemy controller does not initialize the values from function OnStart(), (since game is already started) because of this it just explodes.
Regarding the other issues I did some fixes and changed some more stuff. Thanks for your feedback

ermario commented:
Also found that when using the teleport buttons the character performs the attack (on clicking the button) this will cause the player to not teleport because is on attack animation... Instead if you click on the button while you hold right button (charge shot) , you can teleport without any problem 😂
